### DISCUSSION TITLE
What's the difference between ` "shared": "workspace:*",`  and ` "shared": "workspace:^",` in `package.json`?

### DISCUSSION THREAD

**Discussion by gyhyfj** [pnpm/pnpm#6375] [Q&A] [answered]
When I use `pnpm -F projA add shared` , get  ` "shared": "workspace:^",`  ,  should I change `*` into `^`? 🙏
Another question is when I run `pnpm add shared -wP` at root of my monorepo, `pnpm` will ignore my subrepo named `shared` , is my command incorrect?

**Comment by lvqq** [comment, marked as answer]
The difference is the range. For more details, see: https://pnpm.io/workspaces#publishing-workspace-packages. 

And the command has no problem, it will use exact version in the root of monorepo.

**Reply by gyhyfj** [reply]
Thank you very much
